icefoganalytics/yhsi
# Issue 34: Change Tracking and Approval: Add Management sub-form to Approvibility pathway

Fixes https://github.com/icefoganalytics/yhsi/issues/34

Depends on:
- https://github.com/icefoganalytics/yhsi/pull/32

# User Story
See https://github.com/icefoganalytics/yhsi/issues/3
As an edit level user I can edit the /sites Management form and it will create a site change request for admin approval.

# Details
Now that I have a few working examples, this task is simply a port of the existing patterns to include an additional sub-form.

# Screenshots
![image](https://user-images.githubusercontent.com/23045206/165130668-165d1180-3da0-4c38-85b4-843b65c1e102.png)

# Testing Instructions
1. Update your user roles to include `Site Admin` via the database console.
2. Go to the /sites table and select a site.
3. In the Management form, make an edit and press save.
4. The site will refresh and your edits will be saved.
5.  Update your user roles so that they _do not include_ `Site Admin` or `Administrator` via the database console. Something like `Site Editor` would probably be best.
6. Go to the /sites table and select a site.
7. In the Management form, make and edit (or two) and press save.
8. The page will refresh and all Management fields will become readonly. The save button on the Management form will be disabled with an appropriate tooltip.
9. Make yourself an `Site Admin` again and go to the /sites-change-requests table.
10. The new change request will be at the top of the list. Click on it to review.
11. Note the various features:
   - "accept/reject" toggle buttons
   - accept/reject all
   - save button is locked until all changes are approved/rejected
12. Reject and accept some changes and press Save.
13. Go back to the /sites table and check that you can now edit the site and that the changes have been applied or rejected correctly.
